### ISSUE TITLE
[Bug]: matplotlib plotly style reload library not working with Python3.11

### ISSUE THREAD

**Issue by dimre19** [matplotlib/matplotlib#28980]
### Bug summary

Solved issue 3601 (https://github.com/matplotlib/matplotlib/issues/3601) does not work with Python3.11. Same solution copied, but no effect of reload. matplotlib version is the a actual latest: 3.9.2

### Code for reproduction

```Python
from __future__ import print_function

import os
from contextlib import contextmanager

import matplotlib.pyplot as plt
from matplotlib.style.core import USER_LIBRARY_PATHS


@contextmanager
def temp_style_file(name):
    """ A context manager for creating an empty style file in the expected path.
    """
    filename = os.path.join(USER_LIBRARY_PATHS[0], name)
    with open(filename, 'w'):
        pass
    yield
    os.remove(filename)


print('# styles available:', len(plt.style.available))

with temp_style_file('dummy.mplstyle'):
    print('# before reload:', len(plt.style.available))

    plt.style.reload_library()
    print('# after reload:', len(plt.style.available))
```

### Actual outcome

# styles available: 29
# before reload: 29
# after reload: 29

### Expected outcome

# styles available: 29
# before reload: 29
# after reload: 30

### Additional information

Tested with:
matplotlib 3.9.2
Python3.11

### Operating system

Windows 10

### Matplotlib Version

3.9.2

### Installation

pip

**Comment by ksunden** [comment]
I am not able to reproduce this on my Linux box, so suspect it has to do with Windows-specific conditions.

I will note that I actually didn't have the `<mplconfigdir>/stylelib/` directory created, so it initially errored out entirely, but upon making the directory, it worked as expected, with the number of styles being increased by one.

**Comment by dimre19** [comment]
> I am not able to reproduce this on my Linux box, so suspect it has to do with Windows-specific conditions.
> 
> I will note that I actually didn't have the `<mplconfigdir>/stylelib/` directory created, so it initially errored out entirely, but upon making the directory, it worked as expected, with the number of styles being increased by one.

Yes, you need to create/set up your mpl style library. Glad to hear it works on Linux. Hope someone will test on Windows.
